drivers/platform/stm32/stm32_usb_uart.c

```diff
@@ -0,0 +1,252 @@
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <errno.h>
+#include <stdlib.h>
+#include "no_os_util.h"
+#include "stm32_usb_uart.h"
+#include "no_os_lf256fifo.h"
+#include "no_os_alloc.h"
+#include "usbd_cdc.h"
+
+static uint8_t UserRxBufferFS[2048];
+static uint8_t UserTxBufferFS[2048];
+extern USBD_HandleTypeDef hUsbDeviceFS;
+
+volatile int tx_pending = 0;
+
+/* This global FIFO is needed because of how the STM32 CDC API is implemented.
+ * The function ops do not allow passing context parameters and so it is 
+ * impossible to pass this fifo to CDC_Receive_FS() in any other way.
+ * The implication is that only 1 USB CDC ACM interface may be used at once
+ * but this is an acceptable compromise, since having more than one doesn't
+ * make much sense anyway. */
+struct lf256fifo *gfifo;
+
+static int8_t CDC_Init_FS(void);
+static int8_t CDC_DeInit_FS(void);
+static int8_t CDC_Control_FS(uint8_t cmd, uint8_t* pbuf, uint16_t length);
+static int8_t CDC_Receive_FS(uint8_t* pbuf, uint32_t *Len);
+static int8_t CDC_TransmitCplt_FS(uint8_t *pbuf, uint32_t *Len, uint8_t epnum);
+
+USBD_CDC_ItfTypeDef USBD_Interface_fops_FS = {
+	CDC_Init_FS,
+	CDC_DeInit_FS,
+	CDC_Control_FS,
+	CDC_Receive_FS,
+	CDC_TransmitCplt_FS
+};
+
+static int8_t CDC_Init_FS(void)
+{
+	USBD_CDC_SetTxBuffer(&hUsbDeviceFS, UserTxBufferFS, 0);
```

Comment:
I think we should use snake case for all the functions and variables which are not defined in SDKs.